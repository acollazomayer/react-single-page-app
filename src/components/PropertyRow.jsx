import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Well, Col, ControlLabel } from 'react-bootstrap';

class PropertyRow extends Component {
  render() {
    return (
      <Well bsSize="small">
        <Col xs={3}>
          <ControlLabel>
            {this.props.title}
          </ControlLabel>
        </Col>
        {this.props.message}
      </Well>
    );
  }
}

export default PropertyRow;

PropertyRow.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
};
