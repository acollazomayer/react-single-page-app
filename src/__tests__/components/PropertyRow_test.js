import React from 'react';
import renderer from 'react-test-renderer';
import PropertyRow from '../../components/PropertyRow';

describe('PropertyRow Component', () => {
  const props = {
    title: 'sometitle',
    message: 'somemessage',
  };

  describe('When it renders', () => {
    const wrapper = renderer.create(<PropertyRow {...props} />);

    it('should render with a title and a message', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
