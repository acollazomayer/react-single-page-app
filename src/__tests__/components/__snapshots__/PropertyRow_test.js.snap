// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyRow Component When it renders should render with a title and a message 1`] = `
<div
  className="well well-sm"
>
  <div
    className="col-xs-3"
  >
    <label
      className="control-label"
      htmlFor={undefined}
    >
      sometitle
    </label>
  </div>
  somemessage
</div>
`;
